algolia/algoliasearch-sfcc-b2c
# feat: Handling Dynamic content Values

## Handling Dynamic content Values

Replaced ```$staticlink$, $url(), $httpUrl(), $httpsUrl()``` with SFCC rendered values

## Changes
- Handled dynamic content values for both content asset and page designer pages
- Created a hard-coded configuration
- Wrote unit tests


## How to test

- Enable hardcoded `enableContentFunctionHandler` config in `cartridges/int_algolia/cartridge/scripts/algolia/lib/contentUtil.js`
- Add following content to any content asset and check index on Algolia

```
<h1>---TESTS---</h1>
<div>
    STATIC LINK:
    <p><img width="400" height="400" alt="" src="images/AboutUsBanner/aboutus.jpg?$staticlink$" /> &nbsp;</p <p>
    XXXXimages/AboutUsBanner/aboutus.jpg?$staticlink$XXXX</p>
</div>
<div>
    URL:
    <a title="My URL Search Link" target="_new" href="$Url('Search-ShowContent', 'q', 'customer')$"><span>URL Search
            Link Content Asset Link </span></a>
    <p>XXXXX$Url('Search-ShowContent', 'q', 'customer')$XXXX</p>
</div>
<div>
    HTTP
    XXXXX$httpUrl('Search-Show', 'cgid', 'J','psortb1','category-pos_J')$XXXX
</div>
<div>
    HTTPS
    XXXX$httpsUrl('Search-Show', 'cgid', 'J','psortb1','category-pos_J')$XXXX
</div>
</hr>
```
